core/generated-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/documentation/ExpressionsGivenRow.kt

```diff
@@ -0,0 +1,93 @@
+package org.jetbrains.kotlinx.dataframe.documentation
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.api.*
+import org.jetbrains.kotlinx.dataframe.documentation.ExpressionsGivenRow.*
+import org.jetbrains.kotlinx.dataframe.RowExpression as DfRowExpression
+import org.jetbrains.kotlinx.dataframe.RowValueExpression as DfRowValueExpression
+
+/**
+ * ## Expressions Given Row
+ * Expressing values using a "Row Expression" ([See Row Expressions on the documentation website.](https://kotlin.github.io/dataframe/datarow.html#row-expressions))
+ * can occur in the following two types of operations:
+ *
+ * - Providing a new value for every selected cell given the row of that cell ([Row Expression][org.jetbrains.kotlinx.dataframe.documentation.ExpressionsGivenRow.RowExpression.WithExample]),
+ * for instance in [map][DataFrame.map], [add][DataFrame.add], and [insert][DataFrame.insert]
+ * (using [RowExpression][DfRowExpression]).
+ *
+ * - Providing a new value for every selected cell given the row of that cell and its previous value ([Row Value Expression][org.jetbrains.kotlinx.dataframe.documentation.ExpressionsGivenRow.RowValueExpression.WithExample]),
+ * for instance in [update.with][Update.with], and [convert.notNull][Convert.notNull]
+ * (using [RowValueExpression][DfRowValueExpression]).
+ *
+ * Note:
+ *
+ * [update with][org.jetbrains.kotlinx.dataframe.api.Update.with]- and [add][org.jetbrains.kotlinx.dataframe.api.add]-like expressions use [AddDataRow][org.jetbrains.kotlinx.dataframe.api.AddDataRow] instead of [DataRow][org.jetbrains.kotlinx.dataframe.DataRow] as the DSL's receiver type.
+ * This is an extension to [RowValueExpression][org.jetbrains.kotlinx.dataframe.RowValueExpression] and
+ * [RowExpression][org.jetbrains.kotlinx.dataframe.RowExpression] that provides access to
+ * the modified/generated value of the preceding row ([AddDataRow.newValue][org.jetbrains.kotlinx.dataframe.api.AddDataRow.newValue]).
+ *
+ * A Row Expression is similar to a [Row Condition][org.jetbrains.kotlinx.dataframe.documentation.SelectingRows] but that expects a [Boolean][Boolean] as result.
+ */
+internal interface ExpressionsGivenRow {
+
+    /**
+     * The key for an @arg that will define the operation name for the examples below.
+     * Make sure to [alias][your examples].
```

Comment:
what is the [your examples]?

Reply:
it's nothing, just instructions to the user to alias their examples, while also showing what I mean with aliasing.